cognite/client/utils/_time.py

```diff
@@ -102,6 +106,26 @@ def ms_to_datetime(ms: int | float) -> datetime:
     return datetime(1970, 1, 1, tzinfo=timezone.utc) + timedelta(milliseconds=ms)
 
 
+def datetime_to_ms_iso_timestamp(dt: datetime) -> str:
+    """Converts a datetime object to a string representing a timestamp in the ISO-format expected by the Cognite GraphQL API.
+
+    Args:
+        dt (datetime): Naive or aware datetime object. Naive datetimes are interpreted as local time.
+
+    Raises:
+        TypeError: If dt is not a datetime object
+
+    Returns:
+        str: Timestamp string in ISO 8601 format with milliseconds
+    """
+    if isinstance(dt, datetime):
+        if _datetime_is_tz_naive(dt):
+            return dt.astimezone().isoformat(timespec="milliseconds")
+        return dt.isoformat(timespec="milliseconds")
+    else:
+        raise TypeError(f"Expected datetime object, got {type(dt)}")
```

Comment:
Suggested code golfing 😅 
```suggestion
        if _datetime_is_tz_naive(dt):
            dt = dt.astimezone()
        return dt.isoformat(timespec="milliseconds")
    raise TypeError(f"Expected datetime object, got {type(dt)}")
```

Reply:
Agree. The original reason was testing the tz awareness more easily :)